gui/src/logs.h

```diff
@@ -0,0 +1,57 @@
+#ifndef LOGS_H_
+#define LOGS_H_
+
+#include <QDir>
+#include <QFile>
+#include <QStandardPaths>
+#include <QString>
+#include <QTextStream>
+
+static QString log_path = "";
+
+void logOutputToFile(QtMsgType type, const QMessageLogContext &context,
+                     const QString &msg) {
+  if (log_path.isEmpty()) {
+    // log_path assignment needs to be delayed until first usage of this
+    // function, so it can find the right app folder.
+    log_path =
+        QStandardPaths::writableLocation(QStandardPaths::AppLocalDataLocation);
+#ifdef QT_DEBUG
+    printf("saving logs in %s\n", log_path.toLatin1().data());
+#endif
+    printf("log folder created? %d\n", QDir().mkpath(log_path));
+  }
+  QFile outFile(log_path + "/gui.log");
+  outFile.open(QIODevice::WriteOnly | QIODevice::Append);
```

Comment:
So, I assume this handler gets called every time a log message is written, and then it opens the file anew every time? Is there not a way to open this sink once and have it persist?

Reply:
Could we create and open the log file in the main function before `qInstallMessageHandler` is called? Such as by putting lines 24 and 25 here into the main function and passing `outFile` as a parameter into this function from main?

Reply:
But main doesn't actually call this function, it only provides this this function as a handler. The function gets called by a macro every time there is something like 
```
qCritical() << "Could not serialize GuiStatus";
```

My instinct is that there should be a singleton that persists with the file open. Main would configure this singleton to open a particular file. The function should then address this singleton.

An example of a good singleton implementation here:
https://github.com/ess-dmsc/daquiri/blob/master/source/core/ProducerFactory.h

However, there is the additional problem that this implementation is not thread-safe. The singleton should probably have a mutex.